src/browser/modules/Editor/ProjectFilesButton.tsx

```diff
@@ -0,0 +1,218 @@
+  projectId
+}: ProjectFilesButtonProps): JSX.Element | null => {
+  const [addFile, { error }] = useMutation(ADD_PROJECT_FILE)
+  const [isEditMode, setEditMode] = useState(false)
+  const [isSaveModeClick, setSaveModeClick] = useState(false)
+  const [isProjectFilesMounted, setProjectFilesMounted] = useState(false)
+  const [
+    activeRelateFile,
+    setActiveRelateFile
+  ] = useState<ActiveRelateFile | null>(null)
+
+  useEffect(() => {
+    let isStillMounted = true
+
+    // when a saved Project Script or Local Cache Script is clicked
+    // not sure at this point which it could be
+    bus &&
+      bus.take(EDIT_CONTENT, () => {
+        if (isStillMounted) {
+          setEditMode(false)
+          setActiveRelateFile(null)
+        }
+      })
+    // only when a Project Script is clicked
+    bus &&
+      bus.take(SELECT_PROJECT_FILE, projectCypherFile => {
+        if (isStillMounted) {
+          setEditMode(true)
+          setActiveRelateFile(projectCypherFile)
+        }
+      })
+    // when a non-Project File action sets content in the editor
+    bus &&
+      bus.take(SET_CONTENT, () => {
+        if (isStillMounted) {
+          setEditMode(false)
+          setActiveRelateFile(null)
+        }
+      })
+    // Project Files is mounted
+    bus &&
+      bus.take(PROJECT_FILES_MOUNTED, () => {
+        if (isStillMounted) {
+          setProjectFilesMounted(true)
+        }
+      })
+    // Prject Files is unmounted
+    bus &&
+      bus.take(PROJECT_FILES_UNMOUNTED, () => {
+        if (isStillMounted) {
+          setProjectFilesMounted(false)
+        }
+      })
+
+    return () => {
+      isStillMounted = false
+    }
+  }, [])
+
+  useEffect(() => {
+    // to report an error in the EditorFrame
+    bus && bus.send(PROJECT_FILE_ERROR, '')
+  }, [error])
+
+  useEffect(() => {
+    // bus sends message too quickly before ProjectFiles is mounted,
+    // so need to ensure ProjectsFile is mounted first.
+    // isDrawerOpen is not reliable in this case.
+    if (isSaveModeClick && isProjectFilesMounted) {
+      bus.send(SAVE_PROJECT_FILE, editorValue())
+      setSaveModeClick(false)
+    }
+  })
+
+  if (!isRelateAvailable) return null
+
+  return (
+    <>
+      <EditorButton
+        data-testid={`editor-${PROJECT_EDITOR_BUTTON_TITLE.split(' ')[0]}`}
+        onClick={async () => {
+          if (editorValue().length) {
+            if (isEditMode && activeRelateFile) {
+              // edit mode
+              bus.send(EDIT_PROJECT_FILE_START, '') // display status in the EditorFrame
+              try {
+                const { data } = await addFile({
+                  variables: {
+                    projectId,
+                    fileUpload: new File(
+                      [editorValue()],
+                      activeRelateFile.name
+                    ),
+                    destination: isWindows
+                      ? `${activeRelateFile.directory}\\${activeRelateFile.name}`
```

Comment:
Perhaps not for now but I feel we would need a helper function for building paths, so we don't forget to handle the windows case

Reply:
totally agree. For now, just needed to get this working due to time constraints. I will create a card for this once I am done.

Reply:
https://trello.com/c/8yncUqqQ

Reply:
This might not be needed when sending to the server, we use `path` so it should resolve unix style paths